_episodes/05b-advanced-containers.md

```diff
@@ -0,0 +1,315 @@
+itself, including any of the files on our computer. In order to use Python
+(inside the container) and our script (outside the container, on our computer),
+we need to create a link between the folder on our computer and the container.
+
+This link is called a "mount" and we can create it using an additional
+option to `docker run`. We'll also use the variable `$PWD` which will substitute
+in our current working directory. The option will look like this
+
+`-v $PWD:/temp`
+
+What this means is -- link my current folder with the container, and inside the
+container, name the folder `/temp`
+
+Let's try running the command now:
+~~~
+$ docker run -v $PWD:/temp alice/alpine-python python sum.py
+~~~
+{: .language-bash}
+
+But we get the same error!
+~~~
+python: can't open file 'sum.py': [Errno 2] No such file or directory
+~~~
+{: .output}
+
+This final piece is a bit tricky -- we really have to remember to put ourselves
+inside the container. Where is the `sum.py` file? It's in the folder that's been
+mapped to `/temp` -- so we need to include that in the path to the script. This
+command should give us what we need:
+~~~
+$ docker run -v $PWD:/temp alice/alpine-python python /temp/sum.py
+~~~
+{: .language-bash}
+
+> ## Exercise: Explore the script
+>
+> What happens if you use the `docker run` command above
+> and put numbers after the script name?
+{: .challenge}
+
+> ## Exercise: Checking the options
+>
+> Our Docker command has gotten much longer! Can you go through each piece of
+> the Docker command above the explain what it does? How would you characterize
+> the key components of a Docker command?
+>
+> > ## Solution
+> >
+> > Here's a breakdown of each piece of the command above
+> >
+> > - `docker run`: use Docker to run a container
+> > - `-v $PWD:/temp`: connect my current working directory (`$PWD`) as a folder
+> > inside the container called `/temp`
+> > - `alice/alpine-python`: name of the container to run
+> > - `python /temp/sum.py`: what commands to run in the container
+> >
+> > More generally, every Docker command will have the form:
+> > `docker [action] [docker options] [docker image] [command to run inside]
+> >
+> {: .solution}
+{: .challenge}
+
+> ## Exercise: Interactive jobs
+>
+> Try using the directory mount option but run the container interactively.
+> Can you find the folder that's connected to your computer? What's inside?
+>
+> > ## Solution
+> >
+> > The docker command to run the container interactively is:
+> > ~~~
+> > $ docker run -v $PWD:/temp -it alice/alpine-python sh
+> > ~~~
+> > {: .language-bash}
+> >
+> > Once inside, you should be able to navigate to the `/temp` folder
+> > and see that's contents are the same as the files on your computer:
+> > ~~~
+> > /# cd /temp
+> > /# ls
+> > ~~~
+> > {: .language-bash}
+> {: .solution}
+{: .challenge}
+
+Being able to use files outside the container is very useful when you want to isolate
```

Comment:
I find this paragraph a bit hard to read, esp the "isolate your software in a container" part for some reason.

```suggestion
Mounting a folder can be very useful when you want to run the software inside your container on many different input files.
In other situations, you may want to save or archive an authoritative version of your data by adding it to the container permanently.  That's what we will cover next.
```
Here is an attempt to reword that section.